fix: 4206 - search page misalignment in search-history_view.dart (#4210)

tanishq5414 opened the pull request:
Impacted file:

search-history_view.dart: added padding to the list tile for proper alignment Resolved the misalignment issue on the search page by adding appropriate padding to the list tile in the search-history_view.dart file. This adjustment ensures that the elements within the list are properly aligned, improving the visual consistency of the search page.

### What
- Added padding to the list tile in the `search-history_view.dart` file to achieve proper alignment.
- This adjustment ensures that the elements within the list are properly aligned, improving the visual consistency of the search page.

### Screenshot
![Screenshot_1688317600](https://github.com/openfoodfacts/smooth-app/assets/69450092/b2060942-9347-44d0-9e5a-9022d33efdab)

### Fixes bug(s)
- Fixes: #4206 search page history tiles are misaligned

### Part of 
- #4206

tanishq5414 committed:
fix: 4206 - search page misalignment in search-history_view.dart (#4210)

Impacted file:

search-history_view.dart: added padding to the list tile for proper alignment
Resolved the misalignment issue on the search page by adding appropriate padding to the list tile in the search-history_view.dart file. This adjustment ensures that the elements within the list are properly aligned, improving the visual consistency of the search page.

g123k commented:
Hello, thanks for your PR.
But it seems that it's not perfectly aligned in your screenshot:
![250375133-b2060942-9347-44d0-9e5a-9022d33efdab](https://github.com/openfoodfacts/smooth-app/assets/246838/cc3fad2b-9e7a-4916-9480-8f5cc6b9e520)

g123k reviewed (changes requested):
Furthermore, before pushing, please ensure to format your code.
Thanks

tanishq5414 commented:
Sorry mb, I will make the necessary changes
Does this look good?
![image](https://github.com/openfoodfacts/smooth-app/assets/69450092/235b24cc-1c4c-4520-8ff4-b40006f76448)

g123k commented:
Far better 👍

tanishq5414 committed:
fix: 4206 - search page misalignment in search-history_view.dart (#4210)

Impacted file:

search-history_view.dart: added padding to the list tile for proper alignment
Resolved the misalignment issue on the search page by adding appropriate padding to the list tile in the search-history_view.dart file. This adjustment ensures that the elements within the list are properly aligned, improving the visual consistency of the search page.

tanishq5414 committed:
Merge branch 'develop' of https://github.com/tanishq5414/smooth-app into develop

tanishq5414 commented:
Hey, I have made the changes. Can you verify?

g123k commented:
Almost perfect, the Ripple effect should take the full width:
[Search.webm](https://github.com/openfoodfacts/smooth-app/assets/246838/3f55f3c3-0d6c-45f0-92fd-b6b46fd7d154)

tanishq5414 committed:
fix: Update ripple effect in search-history_view.dart

Refactored the ripple effect in the search-history_view.dart file to enhance the visual feedback when interacting with list tiles

tanishq5414 commented:
Added the ripple effect to the complete List tile. Please verify and let me know if there are any other changes.

g123k commented on packages/smooth_app/lib/pages/scan/search_history_view.dart:
```diff
@@ -49,41 +49,47 @@ class _SearchHistoryViewState extends State<SearchHistoryView> {
+        child: Padding(
+          padding: const EdgeInsets.only(left: 18, right: 21),
+          child: ListTile(
+            contentPadding: EdgeInsets.only(left: 18, right: 21),
```
By curiosity, why do set twice the padding?
Also, by consistency, could you use floating values: 18 -> 18.0

tanishq5414 replied:
Changed the code to follow consistency.
Also, my bad I was testing something and it slipped right through , will be careful before pushing next time

tanishq5414 committed:
fix: Formatted Code

Refactored the code to use floating values in the search-history_view.dart file

tanishq5414 committed:
fix: Formatted Code

Refactored the code to use floating values in the search-history_view.dart file

tanishq5414 commented:
The fix is done ig. It is ready to merge

g123k commented:
It seems that there is still an issue: https://share.waldo.com/294ab6ee641dd9f72e008124c281df26024024?tab=timeline&startTime=39.688

As you can see, the Ripple around the pen (Edit mode) is really small. There should be some padding

tanishq5414 committed:
fixed the ripple effect in edit icon in search_history_view.dart

tanishq5414 commented:
Did the fix!

g123k reviewed (approved):
That's better like so, I will probably implement the accessibility feature later.
Thanks for your PR!
